server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/SearchAction.java

```diff
@@ -0,0 +1,142 @@
+  private final ComponentFinder componentFinder;
+
+  public SearchAction(DbClient dbClient, UserSession userSession, ComponentFinder componentFinder) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentFinder = componentFinder;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_SEARCH)
+      .setDescription(format("List links of a project.<br>" +
+          "The '%s' or '%s' must be provided.<br>" +
+          "Requires one of the following permissions:" +
+          "<ul>" +
+          "<li>'Administer System'</li>" +
+          "<li>'Administer' rights on the specified project</li>" +
+          "<li>'Browse' on the specified project</li>" +
+          "</ul>",
+        PARAM_PROJECT_ID, PARAM_PROJECT_KEY))
+      .setHandler(this)
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setSince("6.1");
+
+    action.createParam(PARAM_PROJECT_ID)
+      .setDescription("Project Id")
+      .setExampleValue(UUID_EXAMPLE_01);
+
+    action.createParam(PARAM_PROJECT_KEY)
+      .setDescription("Project Key")
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    SearchWsRequest searchWsRequest = toSearchWsRequest(request);
+    SearchWsResponse searchWsResponse = doHandle(searchWsRequest);
+
+    writeProtobuf(searchWsResponse, request, response);
+  }
+
+  private SearchWsResponse doHandle(SearchWsRequest searchWsRequest) {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      ComponentDto component = getComponentByUuidOrKey(dbSession, searchWsRequest);
+      List<ComponentLinkDto> links = dbClient.componentLinkDao()
+        .selectByComponentUuid(dbSession, component.uuid());
+      return buildResponse(links);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private SearchWsResponse buildResponse(List<ComponentLinkDto> links) {
+    SearchWsResponse.Builder response = SearchWsResponse.newBuilder();
+    WsProjectLinks.Link.Builder linkBuilder;
+
+    for (ComponentLinkDto link : links) {
+      linkBuilder = WsProjectLinks.Link.newBuilder()
+        .setId(String.valueOf(link.getId()))
+        .setName(link.getName())
+        .setUrl(link.getHref());
+      String type = link.getType();
+      if (type != null) {
+        linkBuilder.setType(type);
+      }
+      response.addLinks(linkBuilder);
+    }
+
+    return response.build();
+  }
+
+  private ComponentDto getComponentByUuidOrKey(DbSession dbSession, SearchWsRequest request) {
+    ComponentDto component = componentFinder.getByUuidOrKey(
+      dbSession,
+      request.getProjectId(),
+      request.getProjectKey(),
+      ComponentFinder.ParamNames.PROJECT_ID_AND_KEY);
+    String projectUuid = firstNonNull(component.projectUuid(), component.uuid());
```

Comment:
no need to get the project uuid anymore. The `userSession.hasComponentUuidPermission(String permission, String componentUuid)` checks the project permission